### PR TITLE
Orchestrate v21.01 update

### PR DESCRIPTION
**Pull request by ggarri**

### CODE REVIEW

File: src/fileRendering.ts [hunk 1]
```diff
@@ -17,14 +18,21 @@ export function copyFilesDir(filesBasePath: string, context: NetworkContext): vo
...
 
         if (!validateDirectoryExists(outputDirname)) {
             fs.mkdirSync(outputDirname, { recursive: true });
         }
 
+        if (isBinaryFileSync(resolvePath(filesBasePath, filePath), size)) {
+            fs.createReadStream(resolvePath(filesBasePath, filePath)).pipe(fs.createWriteStream(outputPath, {
+              mode,
+            }));
+            continue
```
Comment (benjamincburns) [inline src/fileRendering.ts:31]: `continue` and `break` are considered code smells unless you're using them to verify some precondition at the top of a loop. `continue` in this case acts as a sort of trap and creates a secondary exit in the middle of the loop, where a reader who's just scanning through might not expect it. With `if/else` here instead of `continue`, you can more clearly see the control flow and determine the cyclomatic complexity. It also makes the function simpler to edit in the future, as there's only one clear entrance and exit path.

File: src/fileRendering.ts [hunk 2]
```diff
@@ -17,14 +18,21 @@ export function copyFilesDir(filesBasePath: string, context: NetworkContext): vo
...
-        const fileSrc = fs.readFileSync(resolvePath(filesBasePath, filePath), "utf-8");
-        const output = fileSrc.replace(/(\r\n|\n|\r)/gm, os.EOL);
+        const { mode, size } = fs.statSync(resolvePath(filesBasePath, filePath));
 
         if (!validateDirectoryExists(outputDirname)) {
             fs.mkdirSync(outputDirname, { recursive: true });
         }
 
+        if (isBinaryFileSync(resolvePath(filesBasePath, filePath), size)) {
+            fs.createReadStream(resolvePath(filesBasePath, filePath)).pipe(fs.createWriteStream(outputPath, {
```
[the remark this answers is not in the document]
Comment (benjamincburns) [reply src/fileRendering.ts:28]: Also per comments in slack, we should avoid storing executable binaries in this repo, as the intention behind this project is for all executable binaries to be contained and versioned by docker images. This side steps issues around filesystem types and platform compatibility, and gives us a clean encapsulation of the execution environment.

It's fine if you disagree and want to proceed anyway, or if there's a reason why this doesn't work for Orchestrate. In either case just please let me know, as I'd like to understand better the flaws in my strategy here.

File: src/fileRendering.ts (hunk 1 above)
Comment (benjamincburns) [reply src/fileRendering.ts:31]: If you think that I'm being pedantic here, bear in mind that GitHub even uses indented blocks to determine how much context to give you in the diff display when reviewing PRs. For example, have a look at your changes down in `src/networkBuilder.ts` in side-by-side diff mode. See how in the old version it shows you lines 21 and 22, but collapses the branch that you didn't modify?

File: src/networkBuilder.ts [hunk 3]
```diff
@@ -37,44 +38,52 @@ export async function buildNetwork(context: NetworkContext): Promise<void> {
...
         if (context.orchestrate) {
-            console.log(`To start your test network, run 'npm install' and 'npm start' in the installation directory, '${context.outputPath}'`);
+            console.log();
+            console.log(`To start Orchestrate, run 'npm install' and 'npm start' in the directory, '${orchestrateOutputPath}'`);
+            console.log(`For more information on the Orchestrate, see 'README.md' in the directory, '${orchestrateOutputPath}'`);
         } else {
-            console.log(`To start your test network, run 'run.sh' in the installation directory, '${context.outputPath}'`);
+            console.log();
+            console.log(`To start your test network, run 'run.sh' in the directory, '${context.outputPath}'`);
+            console.log(`For more information on the test network, see 'README.md' in the directory, '${context.outputPath}'`);
```
Comment (benjamincburns) [inline src/networkBuilder.ts:85]: What was this change intended to fix? Have you tested it?

Notice that in the old version orchestrate wasn't getting the files in the `common` directory tree, but now it is. Is this intended/desirable behaviour?

File: src/fileRendering.ts (hunk 1 above)
Comment (benjamincburns) [reply src/fileRendering.ts:31]: Okay, it looks like you're using the same `docker-compose` as the other two code paths now. That's good, as it'll allow us to enable private transactions in the orchestrate code path as well.

**Review by benjamincburns — CHANGES_REQUESTED** [review]
Just a couple of small requests for changes.

File: src/networkBuilder.ts (hunk 3 above)
Comment (ggarri) [reply src/networkBuilder.ts:85]: Yes, I've been testing this flow several times locally to validate the installation flow behaves as we intended which means, we want the selected test network (Quorum/Besu) to be loaded along with Orchestrate files.

File: src/fileRendering.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (ggarri) [reply src/fileRendering.ts:28]: Binary file included in Orchestrate files was at the end removed therefore this entire flow is going to be reverted in next version

File: src/fileRendering.ts (hunk 1 above)
Comment (ggarri) [reply src/fileRendering.ts:31]: I don't have any strong preference agains usage of `continue` against  `if/else` statements, although as personal preference I tend to use early exits and loop breaks instead of big if/else block :)

Setting aside code implementation details this code is revered in my next commit as the binary leading to this it is not part of Orchestrate files anymore
